src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -2,46 +2,195 @@
 
 import java.util.List;
 
+@SuppressWarnings("unchecked")
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    class Node<T> {
+        T data;
+        public Node<T> prev = null;
+        public Node<T> next = null;
+
+        public Node(Node<T> prev, T data, Node<T> next) {
+            this.prev = prev;
+            this.data = data;
+            this.next = next;
+        }
+    }
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        addLast(value);
     }
 
-    @Override
     public void add(T value, int index) {
+        indexValidationForAdd(index);
+        if(index == 0) {
+            addFirst(value);
+        }
+        else if(index == size){
+            addLast(value);
+        }
+        else {
+            addToMiddle(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidationForFind(index);
+        Node<T> newNode = head;
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                return newNode.data;
+            }
+            newNode = newNode.next;
+        }
+        throw new IndexOutOfBoundsException();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexValidationForFind(index);
+        Node<T> pointer = head;
+        T oldValue = pointer.data;
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                oldValue = pointer.data;
+                pointer.data = value;
+            }
+            pointer = pointer.next;
+        }
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidationForFind(index);
+        Node<T> newNode = head;
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                removeByNode(newNode);
+            } else {
+                newNode = newNode.next;
+            }
+        }
+        return newNode.data;
```

Comment:
In the `remove(int index)` method, the return statement should be placed after the node is removed. Currently, it returns the data of the node before it is removed, which can lead to incorrect behavior.